apollo-router-core/src/query_planner/selection.rs

```diff
@@ -146,6 +117,32 @@ mod tests {
     use super::*;
     use serde_json::json;
 
+    fn select<'a>(
+        response: &Response,
+        path: &'a Path,
+        selections: &[Selection],
+        schema: &Schema,
+    ) -> Result<Value, FetchError> {
+        let mut values = Vec::new();
+        response
+            .data
+            .select_values_and_paths(path, |_path, value| values.push(value))?;
+
+        Ok(Value::Array(
+            values
+                .into_iter()
+                .flat_map(|value| match (value, selections) {
+                    (Value::Object(content), requires) => {
+                        select_object(content, requires, schema).transpose()
+                    }
+                    (_, _) => Some(Err(FetchError::ExecutionInvalidContent {
+                        reason: "not an object".to_string(),
```

Comment:
❓  it looks like we always have &'static str for ExecutionInvalidContent

Reply:
Unfortunately this type (`FetchError`) is public and should be kept generic. There are a lot of reasons -no pun intended- why we would want to have a `String` for the field `reason`.

Though.... :thinking: One valid alternative would be a `Cow<'static, str>`. This can improve performance if that error happens a lot.

Reply:
Hmm indeed, I don't think we want to optimize that yet, errors shouldn't happen that much should they ?

Reply:
yeah well... not a big deal to change either so :woman_shrugging: